src/org/zaproxy/zap/extension/httpsinfo/MDialog.java

```diff
@@ -0,0 +1,319 @@
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.AbstractFrame;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+public class MDialog extends AbstractFrame {
+	private static final long serialVersionUID = 6633774164518653658L;
+
+	private static final String NEWLINE = "\n";
+	
+	private static final int BEAST_PLUGIN_ID=10200;
+	private static final int CRIME_PLUGIN_ID=10201;
+
+	private ExtensionAlert extensionAlert = (ExtensionAlert) Control.getSingleton().getExtensionLoader()
+			.getExtension(ExtensionAlert.NAME);
+
+	boolean ordered = false;
+	Map<String, Integer> versionStrings;
+	SSLServer server;
+	HttpMessage baseMessage;
+
+	JTextArea general;
+	JLabel spinnerPref;
+	JTextArea cipherSuites;
+	JComboBox<String> suppVers;
+	JButton refresh;
+
+	public MDialog(SSLServer s, HttpMessage msg) {
+		super();
+		this.server = s;
+		this.baseMessage = msg;
+		this.setIconImage(new ImageIcon(ExtensionHttpsInfo.class.getResource(ExtensionHttpsInfo.ICON_PATH)).getImage());
+		this.setTitle(Constant.messages.getString("httpsinfo.name"));
+		this.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+		Dimension d = Toolkit.getDefaultToolkit().getScreenSize();
+		int x = (d.width - getSize().width) / 2;
+		int y = (d.height - getSize().height) / 2;
+		this.setLocation(x, y);
+		init();
+		addComponentsToPane(getContentPane());
+		this.pack();
+		updateContent();
+		this.setVisible(true);
+	}
+
+	public void init() {
+		general = new JTextArea();
+		general.setEditable(false);
+		spinnerPref = new JLabel(Constant.messages.getString("httpsinfo.spinnerpref.label"));
+		cipherSuites = new JTextArea();
+		cipherSuites.setEditable(false);
+		refresh = new JButton(Constant.messages.getString("httpsinfo.serverpref.button"));
+		refresh.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				new Thread(new Runnable() {
+					@Override
+					public void run() {
+						refresh.setEnabled(false);
+						server.orderCertificates();
+						ordered = true;
+						refresh.setVisible(false);
+						updateContent();
+					}
+				}).start();
+			}
+		});
+		versionStrings = new HashMap<String, Integer>();
+		for (int v : server.getSupportedVersions()) {
+			versionStrings.put(SSLServer.versionString(v), v);
+		}
+		Object[] list = versionStrings.keySet().toArray();
+		String[] slist = new String[list.length];
+		for (int i = 0; i < slist.length; i++) {
+			slist[i] = list[i].toString();
+		}
+		Arrays.sort(slist);
+		suppVers = new JComboBox<String>(slist);
+		suppVers.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent arg0) {
+				showCipherSuites();
+
+			}
+		});
+	}
+
+	public void updateContent() {
+		showGeneral();
+		showCipherSuites();
+	}
+
+	public void showGeneral() {
+		StringBuffer content = new StringBuffer(
```

Comment:
Should use StringBuilder instead, the synchronisation does not seem to be needed.